### PR TITLE
Add implementation detail for recommend

### LINKED ISSUE #250: Add recommend implementation to DG
Add implementation details for recommend feature

### PR DESCRIPTION
**Pull request by kevin9foong**
Add implementation detail for recommend 

Issues: 
Resolves #250

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -322,7 +327,74 @@ The implementation of `Model#unlinkFriend()` is as follows:
...
+
+### Recommend feature:
+
+#### Description: 
+
+The recommend feature provides the user a with a command (Format: `recommend -g GAME_ID -t HOUR DAY`) to enter into 
+the gitGud text box which causes gitGud to display a list of friends who are: 
+* available during the specified time (hour and day) based on schedule
+* are linked to the specified game 
+* and are ordered by the highest skill value
```
Comment (tau-bar) [inline docs/DeveloperGuide.md:339]:
```suggestion
The recommend feature provides the user a with a command (Format: `recommend -g GAME_ID -t HOUR DAY`) to enter into 
the gitGud text box which causes gitGud to display a list of friends who are: 
* available during the specified time (hour and day) based on schedule
* linked to the specified game 
* ordered by the highest skill value
```

Just a minor suggestion.

**Review by tau-bar — CHANGES_REQUESTED** [review]
LGTM, minor nits.

**Review by tau-bar — APPROVED** [review]
LGTM, minor nits but I will fix in my PR.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -322,7 +327,74 @@ The implementation of `Model#unlinkFriend()` is as follows:
...
+the new `FriendRecommendFilterPredicate` and `Comparator` instances, which causes filtering and sorting of the list and 
+produces the list of friend recommendations. 
+
+<img src="images/RecommendCommandSequenceDiagram2.png" width="1000" />
+
+<ins>Step 3: Displaying the recommended friends</ins> 
+
+Due to the use of JavaFX's `FilteredList` and `SortedList`, which listens for and tracks for changes whenever the
+`FilteredList#setPredicate(Predicate)` or `SortedList#setComparator(Comparator)` methods are invoked, the user interface is
+notified and updated to display the filtered and sorted friends list based on the produced list of friend recommendations in step 2.
```
Comment (YeluriKetan) [inline docs/DeveloperGuide.md:381]: A bit confusing as to which list the JavaFX-based `UI` is observing. There has been no mention of the term `ObservableList`, which is the main reason the `UI` is being updated whenever there's a change to the underlying `SortedList<Friend>`, which is built on top of the `FilteredList<Friend>`, which in turn is built on top of a `FriendsList`.

**Review by YeluriKetan — COMMENTED** [review]
Already merged but a small issue found.

**Pull request merged**